### PR TITLE
Dismiss keyboard when opening mute chat drawer

### LINKED ISSUE #18485: Keyboard overlays mute chat drawer when opening mute chat drawer from chat
# Problem

Keyboard overlays mute chat drawer when opening mute chat drawer from chat
image

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #18485

### Summary
Dismisses the keyboard when mute chat drawer is opening.

status: ready

### CODE REVIEW

File: src/status_im/contexts/chat/messenger/messages/list/view.cljs [hunk 1]
```diff
@@ -161,6 +161,7 @@
                  :color               cover-bg-color
                  :icon                (if muted? :i/muted :i/activity-center)
                  :on-press            (fn []
+                                        (rf/dispatch [:dismiss-keyboard])
```
Comment (Parveshdhull) [inline src/status_im/contexts/chat/messenger/messages/list/view.cljs]: Hi @ibrkhalil, Thank you very much for finding and fixing this issue.
Please can we add this `:dismiss-keyboard` to `:show-bottom-sheet` event itself?

Comment (ibrkhalil) [reply src/status_im/contexts/chat/messenger/messages/list/view.cljs]: Thanks Parvesh
That's an excellent point, As the bottom sheet and the keyboard won't look nice together in any case.

File: src/status_im/navigation/events.cljs [hunk 2]
```diff
@@ -119,6 +119,7 @@
   {:events [:show-bottom-sheet]}
   [{:keys [db]} content]
   (let [{:keys [sheets hide?]} (:bottom-sheet db)]
+    (rf/dispatch [:dismiss-keyboard])
     (rf/merge {:db (update-in db [:bottom-sheet :sheets] #(conj % content))}
```
Comment (Parveshdhull) [inline src/status_im/navigation/events.cljs]: Please add `dispatch` according to [guideline](https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#registering-event-handlers)

Comment (ibrkhalil) [reply src/status_im/navigation/events.cljs]: Ok

**Review by Parveshdhull — APPROVED** [review]

File: src/status_im/navigation/events.cljs [hunk 3]
```diff
@@ -119,7 +119,8 @@
   {:events [:show-bottom-sheet]}
   [{:keys [db]} content]
   (let [{:keys [sheets hide?]} (:bottom-sheet db)]
-    (rf/merge {:db (update-in db [:bottom-sheet :sheets] #(conj % content))}
+    (rf/merge {:db       (update-in db [:bottom-sheet :sheets] #(conj % content))
+               :dispatch [:dismiss-keyboard]}
```
Comment (clauxx) [inline src/status_im/navigation/events.cljs]: Will it always be the case that we'll want to hide the keyboard when showing the bottom sheet? My assumption is (also) yes, but wondering if there's places that would need the keyboard & bottom sheet together i.e. bottom sheet with an input.

**Review by clauxx — APPROVED** [review]

File: src/status_im/navigation/events.cljs (hunk 3 above)
Comment (ibrkhalil) [reply src/status_im/navigation/events.cljs]: I think the behavior there should be to focus on the input of the bottom sheet once it's open
We're dismissing the keyboard that was there _before_ rendering the bottom sheet.

**Comment by mariia-skrypnyk** [comment]
Hi @ibrkhalil ! Can you please rebase your PR?

**Comment by mariia-skrypnyk** [comment]
Hi @ibrkhalil!

I see such behaviour on Android and iOS and I assume it is an issue you was working on and it still present in this PR.

Android: user can open mute chat drawer but keyboard is still opened (I assume it should be closed and then mute chat drawer opens)

https://github.com/status-im/status-mobile/assets/147824399/c2927234-f1ce-411c-b61d-655a2d6a7e78

iOS: mute chat drawer is still overlapped by the keyboard


https://github.com/status-im/status-mobile/assets/147824399/f3a91cb2-29f7-408d-8fca-c7d05c66e8c4

Please check it on your side and let me know!

**Comment by mariia-skrypnyk** [comment]
Hi @ibrkhalil !

Thanks for your fix.
Failed e2e are not related to your PR. 
So, feel free to merge it.

**Pull request merged**